addons/web_editor/controllers/main.py

```diff
@@ -509,7 +523,11 @@ def modify_image(self, attachment, res_model=None, res_id=None, name=None, data=
             fields['res_id'] = res_id
         if name:
             fields['name'] = name
-        attachment = attachment.copy(fields)
+        if image_optimization_disabled:
+            context = {'image_no_postprocess': True}
+            attachment = attachment.with_context(context).copy(fields)
+        else:
+            attachment = attachment.copy(fields)
```

Comment:
Caller/business code should not have to think about that, just make the endpoint handle that itself and itself only. In this case, why isn't the `_check_contents` doing it itself (not calling _postprocess_contents if ICP set)?